src/02-common-principles.md

```diff
@@ -383,6 +383,200 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## BIDS URI: Pointing to files within and outside of BIDS datasets
+
+Throughout the BIDS specification there a metadata fields that contain one or more
+pointers to files
+(for example `IntendedFor`, `SpatialReference`, and `AssociatedEmptyRoom`).
+Prior to BIDS version 1.7.0, such pointers were often required to be "relative paths".
+However the specification failed to consistently describe relative to *what* these pointers should be specified,
+and the exact syntax to be used.
+Yet even with perfectly well and consistently defined "relative paths",
+this method of pointing to files becomes impractical when referring to files that
+are outside of a particular file tree (for example, on the Internet),
+or when the location of the data that is being pointed to changes.
+
+For these reasons, starting with BIDS version 1.7.0,
+using "relative paths" for all fields pointing to a separate file is [DEPRECATED][].
+For internal consistency, the use of "absolute paths" is [DEPRECATED][] as well.
+
+Instead, the following [URI][uniform-resource-indicator]
+scheme SHOULD be used to point to files within and outside of BIDS datasets,
+as elaborated below.
+
+`bids:<dataset-name>:/absolute/path/within/dataset`
+
+Here, `bids:` is the URI scheme,
+and `<dataset-name>:/absolute/path/within/dataset` is the path to the file,
+consisting of the name of a dataset, a colon,
+and the absolute path within that dataset that MUST start with a forward slash `/`.
+
+The location of a dataset with the name `<dataset-name>` MUST be specified in the
+[`dataset_description.json` file](./03-modality-agnostic-files.md#dataset-description)
+under the `DatasetLinks` field,
+which is an [object][] of [URIs][], as shown in the examples below.
+
+In brief,
+`dataset_description.DatasetLinks` contains mappings from strings to URIs that point to dataset locations.
+The `<dataset-name>`: `""` (that is, an empty string)
+is a reserved value that may only be used to refer to the root of the current dataset;
+an empty string (`""`) MUST NOT be a key in the `DatasetLinks` object.
+
+In the case where a derivatives dataset is nested under a raw dataset and both have a `dataset_description.json` file,
+the BIDS URIs within the nested derivatives dataset MUST be resolved with respect to `/derivatives/dataset_description.json`,
+and the BIDS URIs for the raw dataset (excluding `/derivatives`) MUST be resolved with respect to `/dataset_description.json`.
+
+However if a derivatives dataset is nested under a raw dataset and only the raw dataset has a `dataset_description.json` file,
```

Comment:
This case does not apply: each derivatives dataset (like each dataset) MUST have a `dataset_description.json` file. 

From [here](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#storage-of-derived-datasets): _"In both cases, every derivatives dataset is considered a BIDS dataset and must include a dataset_description.json file at the root level"_

Reply:
Thanks for making me aware of this sentence. I was of the impression that:

- `derivatives/` directories **without** a `dataset_description.json` are just not valid BIDS datasets, but that they can still occur as part of a valid BIDS dataset
- `derivatives/` **without** a `dataset_description.json` thus do **not** get validated by bids-validator
- My interpretation is pretty much the continuation of the status prior to merging the "Common derivatives" proposal into the specification - with the added rule that as soon as you put a `dataset_description.json` into `derivatives/`, it is being validated and must adhere strictly to the BIDS specification on derivatives

Reading the text, my interpretation is not really supported. But realistically, there are still many datasets out there with non-compliant derivatives (from before the time of merging "Common derivatives), and the validator support of derivatives is still non-existent as far as I know.

What do you think of the situation? Should we formalize what I wrote above and put it into the specification? Or should we invalidate all existing datasets with `/derivatives` from before merging "Common derivatives"?

cc @effigies

Reply:
Yes, you are correct: a `/derivatives` _directory_ without `dataset_description.json` is allowed (already starting from v1.0), but it is indeed not a _derivatives dataset_. In the same way the link could also point to a file in `sourcedata`, or to a file in a directory that is ignored (as per `.bidsignore`). 

I think the specific example would become more clear if  it were to point to a file in a directory (either within or outside the top dataset) that is clearly not a BIDS dataset itself (like a template in SPM or AFNI).

Reply:
great, I clarified this in 0c21ae5010c46a2c4d459f3dcf0901f91a10428c

> example would become more clear if it were to point to a file in a directory (either within or outside the top dataset) that is clearly not a BIDS dataset itself (like a template in SPM or AFNI)

I added such examples in other commits and in the sections below this "intro" section